From 5ae643e2ab69892e0dbe296cf14e020c7938ebee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Bornemann?= <joerg.bornemann@qt.io>
Date: Fri, 2 Aug 2019 11:03:06 +0000
Subject: [PATCH] Revert "Yield error if qtquickcompiler is used in non-QML
 projects"

This reverts commit c5578b16d6454e708c8ce12661a85d41eeaaa758, because
it prevents enabling the qtquickcompiler feature globally.

Fixes: QTBUG-77277
Change-Id: Ic80835c462570a67ae3105bb3d1b6452800d2c94
Reviewed-by: Kai Koehne <kai.koehne@qt.io>
---
 tools/qmlcachegen/qtquickcompiler.prf | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/tools/qmlcachegen/qtquickcompiler.prf b/tools/qmlcachegen/qtquickcompiler.prf
index 967d55a5b..b98d8a019 100644
--- a/tools/qmlcachegen/qtquickcompiler.prf
+++ b/tools/qmlcachegen/qtquickcompiler.prf
@@ -1,15 +1,5 @@
 if(qtc_run|lupdate_run): return()
 
-!contains(QT, qml) {
-    qt_modules = \
-        $$replace(QT, -private$, _private) \
-        $$replace(QT_PRIVATE, -private$, _private)
-    qt_modules = $$resolve_depends(qt_modules, "QT.", ".depends" ".run_depends")
-    !contains(qt_modules, qml): \
-        error("The qtquickcompiler feature cannot be used without the QML module.")
-    unset(qt_modules)
-}
-
 qtPrepareTool(QML_CACHEGEN, qmlcachegen, _FILTER)
 qtPrepareTool(QMAKE_RCC, rcc, _DEP)
 
